scripts/quests/windurst/Something_Fishy.lua

```diff
@@ -0,0 +1,98 @@
+-----------------------------------
+-- Something Fishy
+-----------------------------------
+-- Log ID: 2, Quest ID: 52
+-- Tokaka : !pos 240 -189.678 -2 66.335
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.SOMETHING_FISHY)
+
+--quest rewards are handled inside the lua based on repeat turn ins.
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tokaka'] = quest:event(208, 0, xi.items.BASTORE_SARDINE),
+
+            onEventFinish =
+            {
+                [208] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status ~= QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tokaka'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        player:getQuestStatus(quest.areaId, quest.questId) ~= QUEST_AVAILABLE and
+                        npcUtil.tradeHasExactly(trade, xi.items.BASTORE_SARDINE) and
+                        quest:getVar(player, 'Dialog') == 1
+                    then
+                        return quest:event(210, 70, xi.items.BASTORE_SARDINE)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if
+                        player:getQuestStatus(quest.areaId, quest.questId) == QUEST_COMPLETED and
+                        not quest:getMustZone(player) and
+                        quest:getVar(player, 'Dialog') < 1
```

Comment:
There's a lot going on here, and given two different paths, they might be better off broken out into multiple sections (accepted, completed).  There's also no reason to use less than to compare a state that can only be 0 or 1.

Reply:
The issue I ran into is during the capture, after first complete and subsequent "accepting" it doesn't go back into the accepted part of the log. Additionally, you have to get the dialog before trading the fish again after zone. All the cutscenes are the same for first or subsequent turn ins. Spent a decent amount of time trying to get it to match exactly how the capture went. 

As far as the >1 for some reason it would not trigger 208 with not getvar ==1 and there needs to be a discriminator between that and the next one. Up for suggestions to make it cleaner though. Thanks for reviewing!

Reply:
Although I suppose I could do a if quest ~= available and could get rid of those 2 lines...